Extension/c_cpp_properties.schema.json

```diff
@@ -70,9 +70,22 @@
                         ]
                     },
                     "compileCommands": {
-                        "markdownDescription": "Full path to `compile_commands.json` file for the workspace.",
-                        "descriptionHint": "Markdown text between `` should not be translated or localized (they represent literal text) and the capitalization, spacing, and punctuation (including the ``) should not be altered.",
-                        "type": "string"
+                        "oneOf": [
+                            {
+                                "type": "string",
+                                "default": ""
+                            },
+                            {
+                                "type": "array",
+                                "items": {
+                                    "type": "string"
+                                },
+                                "uniqueItems": true,
+                                "default": []
```

Comment:
I believe there should not be any "default" fields in `c_cpp_properties.schema.json` at all, to avoid incorrectly indicating any value as a hard-coded 'default', as the actual defaults may have been overridden using `C_Cpp.default.*` settings.  The one existing occurrence for `enableConfigurationSquiggles` appears to have been a later addition that was in error.  Could you omit these new "default" fields?  (And maybe even remove the "default" from `enableConfigurationSquiggles` as well).

Reply:
The defaults on "C_Cpp.default.compileCommands" are okay though -- my previous comment below was intended to be applied to package.json and not c_cpp_properties.schema.json.

Reply:
@Colengms I added the default values to have some Intellisense because this field has 2 possible types.  
due to the logic in `CppProperties.forceCompileCommandsAsArray`, an empty string, an empty array or an array with empty strings all get demoted to `undefined`.  
This means that value from `C_Cpp.default.compileCommands` will be used instead if valid. (same logic applies to it in `CppSettings`)
![image](https://github.com/user-attachments/assets/c036a702-cb4c-402e-bb7c-7955f3c7d11f)  
@bobbrow was also concerned of this change, so I'm removing it for now. if you change your mind about it I will re-insert this.